peps/tests/test_commands.py

```diff
@@ -6,29 +7,37 @@
 from django.conf import settings
 from django.core import serializers
 from django.core.management import call_command
-from django.core.exceptions import ImproperlyConfigured
 
 import responses
 
 from pages.models import Image
 
-from . import FAKE_PEP_REPO
+from . import FAKE_PEP_ARTIFACT
 
 
-@responses.activate
 class PEPManagementCommandTests(TestCase):
 
-    def setUp(self):
-        self.peps_tar = io.open(FAKE_PEP_ARTIFACT, 'rb')
+    @classmethod
+    def setUpTestData(cls):
+        with io.open(FAKE_PEP_ARTIFACT, 'rb') as fp:
+            peps_tar = fp.read()
         responses.add(
             responses.GET,
             'https://example.net/fake-peps.tar.gz',
             headers={'Last-Modified': 'Sun, 24 Feb 2019 18:01:42 GMT'},
             stream=True,
             content_type='application/x-tar',
             status=200,
-            body=self.peps_tar,
+            body=peps_tar,
         )
+        responses.start()
+        super().setUpTestData()
+
+    @classmethod
+    def tearDownClass(cls):
```

Comment:
I think the previous version (adding `responses.start()` decorator to each test) was better.

Reply:
The previous version (`@responses.activate` decorator) will revert `responses.add` after each test. That make the mock in `setUpTestData`  will only work for the first test method. 

```
test_dump_pep_pages (peps.tests.test_commands.PEPManagementCommandTests) ... ok
test_generate_pep_pages_real (peps.tests.test_commands.PEPManagementCommandTests) ... ERROR
test_image_generated (peps.tests.test_commands.PEPManagementCommandTests) ... ERROR
```

Reply:
Ok, it seems like responses doesn't work well `TestCase` API and the use of `setUpTestData` didn't speed up tests at all (ait's normal because we didn't actually hit the database) Let's use this version:

```diff
diff --git a/peps/tests/test_commands.py b/peps/tests/test_commands.py
index 64606ee..54396b8 100644
--- a/peps/tests/test_commands.py
+++ b/peps/tests/test_commands.py
@@ -1,4 +1,3 @@
-import itertools
 import io
 
 from bs4 import BeautifulSoup
@@ -15,42 +14,35 @@ from pages.models import Image
 from . import FAKE_PEP_ARTIFACT
 
 
+PEP_ARTIFACT_URL = 'https://example.net/fake-peps.tar.gz'
+
+
+@override_settings(PEP_ARTIFACT_URL=PEP_ARTIFACT_URL)
 class PEPManagementCommandTests(TestCase):
 
-    @classmethod
-    def setUpTestData(cls):
-        with io.open(FAKE_PEP_ARTIFACT, 'rb') as fp:
-            peps_tar = fp.read()
+    def setUp(self):
         responses.add(
             responses.GET,
-            'https://example.net/fake-peps.tar.gz',
+            PEP_ARTIFACT_URL,
             headers={'Last-Modified': 'Sun, 24 Feb 2019 18:01:42 GMT'},
             stream=True,
             content_type='application/x-tar',
             status=200,
-            body=peps_tar,
+            body=open(FAKE_PEP_ARTIFACT, 'rb'),
         )
-        responses.start()
-        super().setUpTestData()
-
-    @classmethod
-    def tearDownClass(cls):
-        responses.stop()
-        responses.reset()
-        super().tearDownClass()
 
-    @override_settings(PEP_ARTIFACT_URL='https://example.net/fake-peps.tar.gz')
+    @responses.activate
     def test_generate_pep_pages_real(self):
         call_command('generate_pep_pages')
 
-    @override_settings(PEP_ARTIFACT_URL='https://example.net/fake-peps.tar.gz')
+    @responses.activate
     def test_image_generated(self):
         call_command('generate_pep_pages')
         img = Image.objects.get(page__path='dev/peps/pep-3001/')
         soup = BeautifulSoup(img.page.content.raw, 'lxml')
         self.assertIn(settings.MEDIA_URL, soup.find('img')['src'])
 
-    @override_settings(PEP_ARTIFACT_URL='https://example.net/fake-peps.tar.gz')
+    @responses.activate
     def test_dump_pep_pages(self):
         call_command('generate_pep_pages')
         stdout = io.StringIO()
```

Reply:
👌 Done.